pyrsistent/_pclass.py

```diff
@@ -96,14 +96,18 @@ def set(self, *args, **kwargs):
         """
         if args:
             kwargs[args[0]] = args[1]
+        for key, value in kwargs.items():
+            field = self._pclass_fields.get(key)
+            if field is not None:
+                value = field.factory(value)
```

Comment:
This does not seem to have any effect? I assume you want to set the value for key in kwargs?

Would it be possible to cover the use case in a test?

Reply:
Yeah you were absolutely correct. In suppressing factories for unrelated keys, this mistake skipped them for the keys being set. There's a test for that now as well.